### PR TITLE
Added Vertex.h to globalvertex package

### PR DESCRIPTION
**Pull request by jnclement**
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [ ] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people), EIC, or ECCE (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)
This PR introduces the virtual Vertex base class which will eventually be used as a base for all other vertex types.

## TODOs (if applicable)
Eventually redo inheritance s.t. all vertex classes inherit from this one, and set up vertexmap s.t. all vertices can be accessed from it.

### CODE REVIEW

File: offline/packages/globalvertex/Vertex.h [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  virtual void set_ndof(unsigned int) {}
+
+  virtual float get_position(unsigned int) const { return NAN; }
+  virtual void set_position(unsigned int /*coor*/, float /*xi*/) {}
+
+  virtual float get_error(unsigned int /*i*/, unsigned int /*j*/) const { return NAN; }
+  virtual void set_error(unsigned int /*i*/, unsigned int /*j*/, float /*value*/) {}
+
+  //beam crossing methods
+  virtual float get_beam_crossing() {}
```
Comment (osbornjd) [inline offline/packages/globalvertex/Vertex.h:52]: This should return a dummy float, like `NAN`

File: offline/packages/globalvertex/Vertex.h [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  virtual size_t erase_track(unsigned int /*trackid*/) { return 0; }
+  virtual ConstTrackIter begin_tracks() const;
+  virtual ConstTrackIter find_track(unsigned int trackid) const;
+  virtual ConstTrackIter end_tracks() const;
+  virtual TrackIter begin_tracks();
+  virtual TrackIter find_track(unsigned int trackid);
+  virtual TrackIter end_tracks();
+
+ protected:
+  Vertex() {}
```
Comment (osbornjd) [inline offline/packages/globalvertex/Vertex.h:75]: No constructor needed, as we shouldn't be constructing the base class anyway (only the implemented classes). In fact, that is a mistake in [this](https://github.com/sPHENIX-Collaboration/coresoftware/blob/891e3ab2849000e840d1ee8689545b7adc3c9a63/offline/packages/trackbase_historic/SvtxVertex.h#L76) base class and should be fixed (to do outside of this PR)

File: offline/packages/globalvertex/Vertex.h [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  virtual float get_bbc_q(int) const { return NAN; }
+  virtual float get_bbc_t(int) const { return NAN; }
+
+  //svtxvertex methods
+  virtual void clear_tracks() {}
+  virtual bool empty_tracks() { return true; }
+  virtual size_t size_tracks() const { return 0; }
+  virtual void insert_track(unsigned int /*trackid*/) {}
+  virtual size_t erase_track(unsigned int /*trackid*/) { return 0; }
+  virtual ConstTrackIter begin_tracks() const;
```
[the remark this answers is not in the document]
Comment (jnclement) [reply offline/packages/globalvertex/Vertex.h:67]: Just the begin/find/end, or do we also want clear/empty/size/insert/erase?

**Comment by osbornjd** [comment]
I believe Jin sent out a request to add you to the github collaboration via email. Accept that and then the continuous integration Jenkins checks should start automatically

**Comment by jnclement** [comment]
> I believe Jin sent out a request to add you to the github collaboration via email. Accept that and then the continuous integration Jenkins checks should start automatically

I'm not seeing any such request in my inbox... Spam is empty and I am definitely using the right email for my Github account.

**Review by osbornjd — APPROVED** [review]
Looks good, we'll let Jenkins run now

**Pull request merged**